lib/pure/sugar.nim

```diff
@@ -234,6 +242,21 @@ since (1, 1):
       var s2 = "xyz"
       doAssert s1 & s2 == s1.dup(&= s2)
 
+      proc makePalindrome(s: var string) =
+        for i in countdown(s.len-2, 0):
+          s.add(s[i])
+
+      var c = "xyz"
+
+      # An underscore (_) can be used to denote the place of the argument you're passing:
+      # b = "xyz"
+      var d = dup c:
+        makePalindrome # xyzyx
+        sort(_, SortOrder.Descending) # zyyxx
+        makePalindrome # zyyxxxyyz
+
+      doAssert e == "zyyxxxyyz"
```

Comment:
This appears to be using an undeclared variable

Reply:
My bad, did push without testing locally.